src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -3086,5 +3086,54 @@ private static bool DeclareSameIdentifiers(SyntaxToken[] oldVariables, SyntaxTok
         }
 
         #endregion
+
+        protected override bool IsRudeEditDueToPrimaryConstructor(ISymbol symbol, CancellationToken cancellationToken)
+        {
+            switch (symbol.Kind)
+            {
+                case SymbolKind.NamedType:
+                    break;
+
+                case SymbolKind.Parameter:
+                    {
+                        var container = symbol.ContainingSymbol;
+
+                        if (container is IMethodSymbol { IsImplicitlyDeclared: false, MethodKind: MethodKind.Constructor })
```

Comment:
>if [](http://example.com/codeflow?start=24&length=2)

Is this treating any parameter of an explicit constructor as a rude edit when containing type has a primary constructor? Is that expected for a parameter of a constructor other than the primary constructor?

If this is a temporary restriction, consider adding a `PROTOTYPE` comment.

Reply:
Any edit to a class with primary ctor is a rude edit for now.

Reply:
> Is this treating any parameter of an explicit constructor as a rude edit when containing type has a primary constructor? 

No. Only for a primary constructor.

>Is that expected for a parameter of a constructor other than the primary constructor?

It is not and those are ignored. But as the PR comment says, there is no goal to enable any EnC scenarios in a class/struct with a primary constructor. The goal is to error on the safe side and disable EnC in them. So, if the change were to tread as rude any edit for parameter of a constructor other than the primary constructor, that would be fine.

Reply:
> If this is a temporary restriction, consider adding a PROTOTYPE comment.

The existence of this whole method is temporary. And I have PROTOTYPE comments where it is used. I do not expect it to survive once real EnC support is implemented.